addons/sale_project/models/sale_order.py

```diff
@@ -36,6 +38,14 @@ def _compute_is_product_milestone(self):
         for order in self:
             order.is_product_milestone = order.order_line.product_id.filtered(lambda p: p.service_policy == 'delivered_milestones')
 
+    def _compute_show_project_and_task_button(self):
+        user_is_manager = self.env.user.has_group('project.group_project_manager')
+        user_has_access_right = user_is_manager or self.env.user.has_group('project.group_project_user')
```

Comment:
We can put the groups on the fields instead, so that they are not computed if the user doesn't have the group